### DISCUSSION TITLE
Logout on one device prevents all other devices from getting new jwt

### DISCUSSION THREAD

**Discussion by Vinzent03** [supabase/supabase#3637] [Questions] [answered]
I am using the dart SDK, but I think that doesn't matter.

Let's say I'm signed in on device A and B. Now I log out on Device A. This [revokes](https://github.com/supabase/gotrue#post-logout) all refresh tokens in `auth.refresh_tokens`. Now my jwt expires on device B and I want to get a new jwt by calling the [token](https://github.com/supabase/gotrue#post-token) endpoint. This requires a refresh token, this is saves on the client, but because of the log out on device A, it's revoked and doesn't work. Which causes device B to be logged out too.

So my question is, is this intended or is my workflow wrong?

**Comment by awalias** [comment, marked as answer]
hey @Vinzent03 , flipping this into an issue because I agree we should add support for multiple concurrent sessions as you point out. It shouldn't be too difficult to add, but it will require plenty of testing and a clear migration path (as with all auth changes)

**Reply by awalias** [reply]
https://github.com/supabase/gotrue/issues/248
